### PR TITLE
fix: make baseBranch type non-optional in BranchConfig

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
fix: make baseBranch type non-optional in BranchConfig

## Context
- Ref: #7154

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/repository/update/branch/reuse.spec.ts
Comment (viceice) [inline lib/workers/repository/update/branch/reuse.spec.ts:20]: you should add `baseBranch` to config if it's required by test.

**Comment by RahulGautamSingh** [comment]
Blocked by #17848
After mentioned PR is merged we will use new `BranchCache` with `baseBranch` field. I think better to merge this PR after that one after I update accordingly. 
Okay to not review for now.

**Review by rarkins — DISMISSED** [review]

**Review by viceice — CHANGES_REQUESTED** [review]
blocked

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**
